utilities/it2profile

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+set -o pipefail
+
+# tmux requires unrecognized OSC sequences to be wrapped with DCS tmux;
+# <sequence> ST, and for all ESCs in <sequence> to be replaced with ESC ESC. It
+# only accepts ESC backslash for ST.
+function print_osc() {
+    if [[ $TERM == screen* ]] ; then
+        printf "\033Ptmux;\033\033]"
+    else
+        printf "\033]"
+    fi
+}
+
+# More of the tmux workaround described above.
+function print_st() {
+    if [[ $TERM == screen* ]] ; then
+        printf "\a\033\\"
+    else
+        printf "\a"
+    fi
+}
+
+function get_b64_version() {
+    if [[ -z "${BASE64_VERSION+x}" ]]; then
+        BASE64_VERSION=$(base64 --version 2>&1)
+        export BASE64_VERSION
+    fi
+}
+
+function b64_encode() {
+    get_b64_version
+    if [[ $BASE64_VERSION =~ GNU ]]; then
+        # Disable line wrap
+        base64 -w0
+    else
+        base64
+    fi
+}
+
+function b64_decode() {
+    get_b64_version
+    if [[ $BASE64_VERSION =~ fourmilab ]]; then
+        BASE64_ARG=-d
+    elif [[ $BASE64_VERSION =~ GNU ]]; then
+        BASE64_ARG=-di
+    else
+        BASE64_ARG=-D
+    fi
+    base64 $BASE64_ARG
+}
+
+function error() {
+    echo "ERROR: $*" 1>&2
+}
+
+function errcho() {
+    echo "$@" >&2
+}
+
+function show_help() {
+    errcho
+    errcho "Usage: it2profile -s profile_name"
+    errcho "       it2profile -g"
+    errcho
+    errcho "Change iTerm2 session profile on the fly"
+    errcho
+    errcho "Options:"
+    errcho
+    errcho "    -s      Set current iTerm2 session profile to specified profile name"
+    errcho "    -g      Get current iTerm2 session profile name"
+    errcho "    -r      Reset to the session initial profile"
+    errcho
+}
+
+function check_dependency() {
+    if ! (builtin command -V "$1" >/dev/null 2>&1); then
+        error "missing dependency: can't find $1"
```

Comment:
errcho

Reply:
We're should not confuse them, because `errcho()` is about output to STDERR and `error()` is for printing an `ERROR:` message. This is convenient to separate them in case if we need to add option to suppress printing of all error messages. Of course we can reuse `errcho()` in `error()`.

Reply:
Errors should be printed to stderr, whether it is via `error` or `errcho`